src/core/core.scale.js

```diff
@@ -957,34 +963,66 @@ var Scale = Element.extend({
 				}
 			}
 
-			items.push({
-				tx1: tx1,
-				ty1: ty1,
-				tx2: tx2,
-				ty2: ty2,
-				x1: x1,
-				y1: y1,
-				x2: x2,
-				y2: y2,
-				labelX: labelX,
-				labelY: labelY,
-				glWidth: lineWidth,
-				glColor: lineColor,
-				glBorderDash: borderDash,
-				glBorderDashOffset: borderDashOffset,
-				rotation: -1 * labelRotationRadians,
+			if (lineValue !== undefined) {
+				gridLineItems.push({
+					tx1: tx1,
+					ty1: ty1,
+					tx2: tx2,
+					ty2: ty2,
+					x1: x1,
+					y1: y1,
+					x2: x2,
+					y2: y2,
+					width: lineWidth,
+					color: lineColor,
+					borderDash: borderDash,
+					borderDashOffset: borderDashOffset,
+				});
+			}
+
+			if (index === ticks.length - 1 && offsetGridLines) {
```

Comment:
This seems to have a lot of duplication. Can we change `helpers.each(ticks, function(tick, index) {` to `for (i = 0; i < ticks.length + (offsetGridLines ? 1 : 0); i++)` ?

Reply:
I removed the duplication by adding an extra tick to `ticks`.